packages/jupyter-ai-magics/jupyter_ai_magics/magics.py

```diff
@@ -355,14 +366,8 @@ def _append_exchange(self, prompt: str, output: str):
 
         TODO: bound this list to length `self.max_history * 2`.
         """
-        self.transcript.append({
-            "role": "user",
-            "content": prompt
-        })
-        self.transcript.append({
-            "role": "assistant",
-            "content": output
-        })
+        self.transcript.append({"role": "user", "content": prompt})
+        self.transcript.append({"role": "assistant", "content": output})
```

Comment:
Can you bound this list to be of length `self.max_history * 2`? Right now, the `ai()` method is manually truncating `self.transcript` every time it appends it to the chat history. It would be more reliable to add some logic to bound the length of `self.transcript` in the `_append_exchange()` method itself.